test/e2e/storage/volume_provisioning.go

```diff
@@ -163,6 +163,80 @@ func testDynamicProvisioning(t storageClassTest, client clientset.Interface, cla
 	return pv
 }
 
+// testGlusterDynamicProvisioning tests the same workflow of testDynamicProvisioning expect it wont attach
+// pv to a pod volume and try write/read. This is due to a limitation in heketi which require raw devices to
+// provision a real volume. But this test make sure heketi responded properly for the volumecreate requests
+// from the gluster dynamic provisioner.
+func testGlusterDynamicProvisioning(t storageClassTest, client clientset.Interface, claim *v1.PersistentVolumeClaim, class *storage.StorageClass) {
+	var err error
+	if class != nil {
+		By("creating a StorageClass " + class.Name)
+		class, err = client.StorageV1().StorageClasses().Create(class)
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			framework.Logf("deleting storage class %s", class.Name)
+			framework.ExpectNoError(client.StorageV1().StorageClasses().Delete(class.Name, nil))
+		}()
+	}
+
+	By("creating a claim")
+	claim, err = client.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(claim)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		framework.Logf("deleting claim %q/%q", claim.Namespace, claim.Name)
+		// typically this claim has already been deleted
+		err = client.CoreV1().PersistentVolumeClaims(claim.Namespace).Delete(claim.Name, nil)
```

Comment:
there is a helper [func](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pv_util.go#L161) to delete a pvc and check for err IsNotFound.

Reply:
yes, used the mentioned helper function.